phpstan.el

```diff
@@ -28,19 +28,83 @@
 ;; https://github.com/phpstan/phpstan
 
 ;;; Code:
+(require 'php-project)
 (require 'flycheck nil)
 
+
+;; Variables:
+
+;;;###autoload
+(progn
+  (defvar phpstan-config-file nil)
```

Comment:
What's the reason of having these as variables instead of flycheck options?  The flycheck macro expands to a defcustom so we can still change it manually just as a defvar but it probides good documentation and listing of options for users.